airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/stream_slicer.py

```diff
@@ -23,24 +23,10 @@ class StreamSlicer(RequestOptionsProvider):
     """
 
     @abstractmethod
-    def stream_slices(self, sync_mode: SyncMode, stream_state: StreamState) -> Iterable[StreamSlice]:
+    def stream_slices(self, sync_mode: SyncMode) -> Iterable[StreamSlice]:
```

Comment:
I think we can probably remove `sync_mode` from the interface as well but we should probably answer the questions in the description in `User Impact -> SubstreamPartitionRouter`

Reply:
After [discussing this internally](https://airbytehq-team.slack.com/archives/C02TL38U5L7/p1686939006489719), we decided to remove the `sync_mode` from the interface of this method